src/Profile/Shopware/Converter/CategoryConverter.php

```diff
@@ -84,19 +82,11 @@ public function convert(
         $this->connectionId = $connection->getId();
         $this->connectionName = $connection->getName();
 
-        if (!isset($data['_locale'])) {
-            $this->loggingService->addLogEntry(
-                SwagMigrationLogBuilder::fromMigrationContext($migrationContext)
-                    ->withEntityName(CategoryDefinition::ENTITY_NAME)
-                    ->withFieldSourcePath('_locale')
-                    ->withSourceData($data)
-                    ->build(EmptyNecessaryFieldRunLog::class)
-            );
+        $converted = [];
 
-            return new ConvertStruct(null, $data);
+        if (isset($data['_locale'])) {
+            $this->locale = $data['_locale'];
         }
```

Comment:
I would add an `else` branch here and set `$this->locale` to `null`. Otherwise the value could stay for the next call to convert, which might cause a bug that is really hard to reason about.

Previously this wasn't an issue because of the early return and always setting it to the value of `$data['_locale']`, which after the early return always exists.